src/platform_impl/windows/event_loop.rs

```diff
@@ -2004,26 +2004,23 @@ unsafe fn public_window_callback_inner<T: 'static>(
         }
 
         WM_SETCURSOR => {
-            let set_cursor_to = {
-                let window_state = userdata.window_state_lock();
-                // The return value for the preceding `WM_NCHITTEST` message is conveniently
-                // provided through the low-order word of lParam. We use that here since
-                // `WM_MOUSEMOVE` seems to come after `WM_SETCURSOR` for a given cursor movement.
-                let in_client_area = super::loword(lparam as u32) as u32 == HTCLIENT;
-                if in_client_area {
-                    Some(window_state.mouse.cursor)
-                } else {
-                    None
-                }
-            };
-
-            match set_cursor_to {
-                Some(cursor) => {
-                    let cursor = unsafe { LoadCursorW(0, util::to_windows_cursor(cursor)) };
-                    unsafe { SetCursor(cursor) };
-                    result = ProcResult::Value(0);
-                }
-                None => result = ProcResult::DefWindowProc(wparam),
+            let window_state = userdata.window_state_lock();
```

Comment:
we should keep locks to a minimum, in particular not covering a call to `DefWindowProc(wparam)` as it might be re-entrant